test/conf/nginx.conf

```diff
@@ -153,6 +153,19 @@ http {
         }
     }
 
+    server {
+        listen 58089 ssl;
+        server_name localhost;
+        ssl_protocols       TLSv1 TLSv1.1 TLSv1.2;
+        ssl_ciphers         HIGH:!aNULL:!MD5;
+        ssl_certificate     __NGXDOCROOT__/dummy.crt;
+        ssl_certificate_key __NGXDOCROOT__/dummy.key;
+        ssl_verify_client on;
+        ssl_client_certificate __NGXDOCROOT__/dummy.crt;
+
+        mruby_ssl_verify_client_handler_code 'Nginx::SSL.reject_client';
```

Comment:
Could you add more test like accept_client method? If possible, please also write a test that indicates that there is no problem even if combined with ssl_handshake_handler.

Reply:
OpenSSL 1.0.2g  fail the test.

```
Fail: ngx_mruby - ngx_mruby_ssl_verify_client_handler with Nginx::SSL.reject_client
 - Assertion[1] Failed: Expected to be equal
    Expected: 35
      Actual: 0
Total: 82
   OK: 81
   KO: 1
Crash: 0
 Time: 2.401487 seconds

trace (most recent call last):
```

- error.log

```
2018/02/04 13:28:27 [info] 31317#0: *105 client sent no required SSL certificate while reading client request headers, client: 127.0.0.1, server: localhost, request: "GET / HTTP/1.1", host: "localhost:58089"
```

Reply:
Thanks for your comments and report.
I would realize that test by curl is very boring work.
so  I'd like to use `openssl s_client` for `Nginx::SSL.(accept|reject)_client`.
It is useful to check whether SSL handshake is success or not and 
testing for them does not need HTTP request/response.

> If possible, please also write a test that indicates that there is no problem even if combined with ssl_handshake_handler.
I'm sure but I have a question.
What does `no problem` mean?
I would guess each directive can run vaild ruby (like "a= 1+1") code in SSL Handshake, you mean?

Reply:
> each directive can run vaild ruby (like "a= 1+1") code in SSL Handshake

Yes, `mrb_state` of verify_handler shared between `mscf->state->mrb` of other ssl_handler state. So I want to test about the mrb_state.